environment-api/src/test/java/com/sequenceiq/environment/api/v1/environment/model/response/EnvironmentStatusTest.java

```diff
@@ -72,6 +82,17 @@ void isAvailableTest(String testCaseName, EnvironmentStatus underTest, boolean a
                 { "UPDATE_FAILED",                          EnvironmentStatus.UPDATE_FAILED,                        true },
                 { "ENVIRONMENT_RESOURCE_ENCRYPTION_INITIALIZATION_IN_PROGRESS", EnvironmentStatus.ENVIRONMENT_RESOURCE_ENCRYPTION_INITIALIZATION_IN_PROGRESS, false },
                 { "ENVIRONMENT_RESOURCE_ENCRYPTION_DELETE_IN_PROGRESS", EnvironmentStatus.ENVIRONMENT_RESOURCE_ENCRYPTION_DELETE_IN_PROGRESS, false },
+                { "FREEIPA_DELETED_ON_PROVIDER_SIDE",       EnvironmentStatus.FREEIPA_DELETED_ON_PROVIDER_SIDE,     true },
+                { "UPGRADE_CCM_VALIDATION_IN_PROGRESS",     EnvironmentStatus.UPGRADE_CCM_VALIDATION_IN_PROGRESS,   false },
+                { "UPGRADE_CCM_ON_FREEIPA_IN_PROGRESS",     EnvironmentStatus.UPGRADE_CCM_ON_FREEIPA_IN_PROGRESS,   false },
+                { "UPGRADE_CCM_ON_DATALAKE_IN_PROGRESS",    EnvironmentStatus.UPGRADE_CCM_ON_DATALAKE_IN_PROGRESS,  false },
+                { "UPGRADE_CCM_ON_DATAHUB_IN_PROGRESS",     EnvironmentStatus.UPGRADE_CCM_ON_DATAHUB_IN_PROGRESS,   false },
+                { "UPGRADE_CCM_ROLLING_BACK",               EnvironmentStatus.UPGRADE_CCM_ROLLING_BACK,             false },
```

Comment:
I wonder if the `UPGRADE_CCM_ROLLING_BACK` state should be considered a failed one. But perhaps not, I suppose the transition will look like e.g. `UPGRADE_CCM_ON_FREEIPA_IN_PROGRESS` -> `UPGRADE_CCM_ROLLING_BACK` -> `UPGRADE_CCM_ON_FREEIPA_FAILED`, right?

Reply:
this is not yet planned but yes it seems a good idea.